src/react/NewPipelinePage.js

```diff
@@ -16,145 +16,159 @@
-        && pipelineName != "new"
-        && pipelineSource != ""
-    );
-  }
-
-  onNameChanged(e) {
-    const newVal = e.target.value;
-    this.setState({
-        pipelineName: newVal
-    });
-  }
-
-  onSourceChanged(e) {
-    const newVal = e.target.value;
-    this.setState({
-        pipelineSource: newVal
-    });
-  }
-
-  onDescriptionChanged(e) {
-    const newVal = e.target.value;
-    this.setState({
-      pipelineDescription: newVal
-    });
-  }
-
-  render() {
-    const { alerts, pipelineName, pipelineSource, pipelineDescription } = this.state;
-
-    return (
-      <SingleCardPage
-        title="New Pipeline"
-        errorMessages={alerts}
-        onAlertClose={this.closeAlert}
+  }, [astarte, history, setIsCreatingPipeline, addAlert, pipelineName, pipelineSource, schemaObject, pipelineDescription]);
+
+  const schemaObject = useMemo(() => {
+    if (pipelineSchema === '') {
+      return false;
+    }
+    try {
+      const schema = JSON.parse(pipelineSchema);
+      return schema;
+    } catch (e) {
+      return undefined;
+    }
+  }, [pipelineSchema]);
+
+  const isValidSchema = useMemo(() => {
+    if (!schemaObject) {
+      return false;
+    }
+    try {
+      ajv.compile(schemaObject);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }, [schemaObject, ajv]);
+
+  return (
+    <SingleCardPage
+      title="New Pipeline"
+      errorMessages={alerts}
+      onAlertClose={closeAlert}
+    >
+      <Form>
+        <Form.Group controlId="pipeline-name">
+          <Form.Label>Name</Form.Label>
+          <Form.Control
+            type="text"
+            value={pipelineName}
+            onChange={(e) => setPipelineName(e.target.value)}
+          />
+        </Form.Group>
+        <Form.Group controlId="pipeline-source">
+          <Form.Label>Source</Form.Label>
+          <Form.Control
+            as="textarea"
+            rows={8}
+            value={pipelineSource}
+            onChange={(e) => setPipelineSource(e.target.value)}
+          />
+        </Form.Group>
+        <Form.Group controlId="pipeline-schema">
+          <Form.Label>Schema</Form.Label>
+          <Form.Control
+            as="textarea"
+            rows={8}
+            value={pipelineSchema}
+            isValid={pipelineSchema !== '' && isValidSchema}
+            isInvalid={pipelineSchema !== '' && !isValidSchema}
+            onChange={(e) => setPipelineSchema(e.target.value)}
+          />
+        </Form.Group>
+        <Form.Group controlId="pipeline-description">
+          <Form.Label>Description</Form.Label>
+          <Form.Control
+            as="textarea"
+            value={pipelineDescription}
+            onChange={(e) => setPipelineDescription(e.target.value)}
+          />
+        </Form.Group>
+      </Form>
+      <Button
+        variant="primary"
+        onClick={createPipeline}
+        disabled={pipelineName === '' || pipelineName === 'new' || pipelineSource === ''}
```

Comment:
add `isCreatingPipeline` to disabled condition

Reply:
You could define a `isValidForm` to keep this line cleaner